cli/script/command-executor.ts

```diff
@@ -840,8 +841,8 @@ function promote(command: cli.IPromoteCommand): Promise<void> {
 export var release = (command: cli.IReleaseCommand): Promise<void> => {
     if (isBinaryOrZip(command.package)) {
         throw new Error("It is unnecessary to package releases in a .zip or binary file. Please specify the direct path to the update content's directory (e.g. /platforms/ios/www) or file (e.g. main.jsbundle).");
-    } else if (semver.valid(command.appStoreVersion) === null) {
-        throw new Error("Please use a semver-compliant app store version, for example \"1.0.3\".");
+    } else if (semver.validRange(command.appStoreVersion) === null) {
+        throw new Error("Please use a semver-compliant target binary version range, for example \"1.0.0\", \"*\" or \"^1.2.3\".");
```

Comment:
I don't think we need to optimize the performance in error cases like this. More importantly, I think client-side validation comes with a maintenance cost, as well as the cost of having less flexibility to change things on the server (which we've already run into a few times).

I agree with having a friendlier error message, and ideally that would be generated on the server, but I understand if it's too much work to do it as part of this PR.